src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -0,0 +1,230 @@
+import re
+import time
+import sys
+from datetime import datetime, timedelta
+from io import BytesIO
+import time
+import colorama
+from azure.common import AzureHttpError
+from azure.storage.blob import AppendBlobService, ContainerPermissions
+from .sdk.models import (
+    QueueBuildRequest,
+    DockerBuildParameters,
+    PlatformProperties,
+    Build,
+    BuildArgument
+)
+from azure.cli.core.commands import LongRunningOperation
+import pytz
+
+
+def acr_show_logs(cmd, client, registry_name, build_id, resource_group_name=None):
+    # Required and hardcoded for now. Refer to the "HACK" introduced in build_operations.py
+    custom_headers = dict()
+    custom_headers['logType'] = 'RawText'
+    buildLogResult = client.get_log_link(
+        build_id=build_id, resource_group_name=resource_group_name, registry_name=registry_name, custom_headers=custom_headers)
+    logFileSas = buildLogResult.log_link
+
+    if not logFileSas:
+        return 'No logs found.'
+
+    match = get_match(logFileSas)
+    accountName = match.group('accountName')
+    containerName = match.group('containerName')
+    blobName = match.group('blobName')
+    sasToken = match.group('sasToken')
+
+    if not accountName:
+        raise ValueError(
+            "Unable to parse the account name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not containerName:
+        raise ValueError(
+            "Unable to parse the container name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not blobName:
+        raise ValueError(
+            "Unable to parse the blob name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not sasToken:
+        raise ValueError(
+            "Unable to parse the SAS token from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
```

Comment:
Dunno about that one, leaving this as-is for now. The CLI guys can complain if it's a big deal to raise errors. In this case, if we parse incorrectly (could be a server bug (bad token) OR a client bug (bad regex)), we end up with a 404 from blob. We could also end up with a 404 from blob in a lot of other situations, e.g., the log file hasn't been created yet and we want to retry. In this case it's an absolute failure from the get-go, in cases they're potentially transient.

Reply:
It is arguable, IMO exceptions should be caught rather than thrown to users. Users probably don't care what/why a command is not working if it's not working and even if they care the root cause, the exceptions can hardly provide useful information to them. I think the goal here is to have no client bug (i.e., throwing exceptions) while server bug should be handled gracefully.